lib/util.py

```diff
@@ -92,44 +28,40 @@ def wrapper(command):
             data = {
                 "name": parameter.name,
                 "required": parameter.default == inspect.Parameter.empty,
-                "option_type": get_option_type(parameter.annotation),
+                "type": interactions.OptionType.from_type(parameter.annotation),
                 "description": f'{parameter.name} (missing description)'
             }
 
             merged_data = {**data, **kwargs.get(parameter.name, {})}
+            kwargs.pop(parameter.name, None)
             options.append(merged_data)
 
         options = [
-            create_option(**option)
+            interactions.SlashCommandOption(**option)
             for option in options
         ] if len(options) > 0 else None
-        help_wrapper = cog_ext.cog_slash(
-            name=command.__name__,
-            guild_ids=GUILD_IDS,
-            description=command.__doc__,
-            options=options,
-        )
-
-        return help_wrapper(command)
-
-    return wrapper
+        commandArgs = {
+            "name": sanitize_name(command.__name__),
+            "scopes": GUILD_IDS,
+            "description": command.__doc__,
+            "options": options,
+            **kwargs,
+        }
+        help_wrapper = interactions.slash_command(**commandArgs)
+        wrapped = help_wrapper(command)
+
+        # inherit attributes set by other decorators
+        for attr in dir(command):
+            if (not attr.startswith("_") and (not hasattr(wrapped, attr) or getattr(wrapped, attr) == None)):
```

Comment:
this if statement is extremely confusing. Can you break this up into 2 fail cases:

```python3
if attr.startswith("_"):
   continue
 if hasattr(wrapped, attr) and getattr(wrapped, attr) is not None:
     continue
  ```